src/test/js/App.test.js

```diff
@@ -81,9 +105,74 @@ describe('Test adding questions', () => {
                 case 'numberQuestion':
                     sumNumber += 1;
                     break;
+                case 'dropdown':
+                    sumDrop += 1;
+                    break;
             }
             expect(wrapper.find({ label: "Open Question Input" }).length).toBe(sumOpen);
             expect(wrapper.find({ label: "Number Question Input" }).length).toBe(sumNumber);
+            expect(wrapper.find({ label: "Dropdown Question Input" }).length).toBe(sumDrop);
         }
    });
+
+   beforeEach(() => {
+        fetch.resetMocks()
+   });
+
+   it('Save survey', (done) => {
+
+        let createRequest;
+        let nameRequest;
+        let createPromise;
+
+        // uses 'jest-fetch-mock'
+        fetch.mockResponse(req => {
+            if (req.url.endsWith('createSurvey')) {
+                createRequest = req;
+                createPromise = Promise.resolve({
+                    status: 200,
+                    body  : JSON.stringify({ message : 'ok', id : 1 }),
+                    json: () => Promise.resolve({ message : 'ok', id : 1 }),
+                });
+                return createPromise;
+            } else if (req.url.endsWith("retrieveSurveyNames")) {
+                return Promise.resolve({
+                    status: 200,
+                    body  : JSON.stringify({ nameList : ["Survey1"], idList : [1] }),
+                    json: () => Promise.resolve({ nameList : ["Survey1"], idList : [1] }),
+                });
+            } else {
+                return Promise.resolve({
+                    status: 400,
```

Comment:
Our fetch mock should throw 404 if the URL cannot be found. It will make debugging tests easier in the future.